joplin/api/schema.py

```diff
@@ -119,10 +119,14 @@ class Meta:
 
 class PageRevisionNode(DjangoObjectType):
     as_service_page = graphene.NonNull(ServicePageNode)
+    as_process_page = graphene.NonNull(ProcessPageNode)
 
     def resolve_as_service_page(self, resolve_info, *args, **kwargs):
         return self.as_page_object();
 
+    def resolve_as_process_page(self, resolve_info, *args, **kwargs):
```

Comment:
This seems like it might get unwieldy overtime. Is there a path to refactor this towards `resolve_as_page` and get the page type via attributes?

Reply:
I would love to make that possible, however, as_page_object() returns either a ServicePageNode or a ProcessPageNode. I tried abstracting it and using a PageNode before but I ran into a type error.

Overall I'm not super concerned about it, we already are making different graphQL queries based on the different properties we want for any given page.